torchaudio/datasets/speechcommands.py

```diff
@@ -81,6 +92,26 @@ def __init__(self,
 
         walker = walk_files(self._path, suffix=".wav", prefix=True)
         walker = filter(lambda w: HASH_DIVIDER in w and EXCEPT_FOLDER not in w, walker)
+
+        if subset in ["training", "validation"]:
+            filepath = os.path.join(self._path, "validation_list.txt")
+            with open(filepath) as f:
+                validation_list = [os.path.join(self._path, l.strip()) for l in f.readlines()]
+
+        if subset in ["training", "testing"]:
+            filepath = os.path.join(self._path, "testing_list.txt")
+            with open(filepath) as f:
+                testing_list = [os.path.join(self._path, l.strip()) for l in f.readlines()]
+
+        if subset == "validation":
+            walker = validation_list
+        elif subset == "testing":
+            walker = testing_list
+        elif subset == "training":
+            walker = filter(
+                lambda w: not (w in validation_list or w in testing_list), walker
+            )
+
         self._walker = list(walker)
```

Comment:
`list` is no longer necessary and should be removed.

Reply:
same

Reply:
With the change mentioned [above](https://github.com/pytorch/audio/pull/966/#discussion_r512198533), this is indeed no longer needed. Removed.